src/plugins/diagram-viewer/diagram-content.test.ts

```diff
@@ -0,0 +1,125 @@
+import { castToSnapshot, isAlive, types } from "mobx-state-tree";
+import { when } from "mobx";
+import { defaultDiagramContent, DiagramContentModel } from "./diagram-content";
+
+const TestContainer = types.model("TestContainer", {
+  content: DiagramContentModel
+});
+
+describe("DiagramContent", () => {
+  it("has default content", () => {
+    const content = defaultDiagramContent();
+    expect(content.root).toBeDefined();
+  });
+
+  it("can export content", () => {
+    const content = DiagramContentModel.create();
+    const expected = JSON.stringify({nodes: {}});
+    expect(content.exportJson()).toEqual(expected);
+  });
+
+  it("is always user resizable", () => {
+    const content = DiagramContentModel.create();
+    expect(content.isUserResizable).toBe(true);
+  });
+
+  it("setups up variables api after being attached", () => {
+    const content = DiagramContentModel.create();
+    const container = TestContainer.create({content: castToSnapshot(content)});
+    expect(container.content.root.variablesAPI).toBeDefined();
+  });
+
+  // This is more of an integration test because it is testing the innards of 
+  // DQRoot, but it exercises code provided by the diagram-content in the process
+  it("supports creating Nodes", () => {
+    const content = DiagramContentModel.create();
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    const container = TestContainer.create({content: castToSnapshot(content)});
+
+    expect(content.root.nodes.size).toBe(0);
+    expect(content.sharedModel?.variables.length).toBe(0);
+
+    content.root.createNode({x: 1, y: 1});
+    expect(content.root.nodes.size).toBe(1);
+    const newNode = Array.from(content.root.nodes.values())[0]; 
+    if (!newNode) {
+      throw new Error("can't find node that was just added");
+    }
```

Comment:
The reason is because Jest's expectation aren't compatible with Typescript's type checking. I put a comment about this an earlier PR:

>       // We are throwing here instead of using an expect, so typescript picks
>       // this up as a type guard. 
>       // Perhaps in the future Jest's expect statements can act as assertions
>       // https://github.com/DefinitelyTyped/DefinitelyTyped/issues/41179

So by doing the check and then throwing we can assume the newNode is valid and don't have to use `?.` when dereferencing.

I'll try to make a utility function that works with Typescript and that makes this clear.

Reply:
I guess I didn't completely grok the prior comment but now it makes more sense.

Reply:
I've got a utility function that I'll bring in with a different PR so it is easier to show to other developers incase they want to use it in their projects.

Reply:
https://github.com/concord-consortium/collaborative-learning/pull/1253